cartoframes/data/dataset.py

```diff
@@ -515,49 +497,5 @@ def _get_geom_col_type(df):
     return geom.geom_type
 
 
-def _first_not_null_value(df, col):
-    return df[col].loc[~df[col].isnull()].iloc[0]
-
-
-def _encode_decode_decorator(func):
-    """decorator for encoding and decoding geoms"""
-    def wrapper(*args):
-        """error catching"""
-        try:
-            processed_geom = func(*args)
-            return processed_geom
-        except ImportError as err:
-            raise ImportError('The Python package `shapely` needs to be '
-                              'installed to encode or decode geometries. '
-                              '({})'.format(err))
-    return wrapper
-
-
-@_encode_decode_decorator
-def _decode_geom(ewkb):
-    """Decode encoded wkb into a shapely geometry
-    """
-    # it's already a shapely object
-    if hasattr(ewkb, 'geom_type'):
-        return ewkb
-
-    from shapely import wkb
-    from shapely import wkt
-    if ewkb:
-        try:
-            return wkb.loads(ba.unhexlify(ewkb))
-        except Exception:
-            try:
-                return wkb.loads(ba.unhexlify(ewkb), hex=True)
-            except Exception:
-                try:
-                    return wkb.loads(ewkb, hex=True)
-                except Exception:
-                    try:
-                        return wkb.loads(ewkb)
-                    except Exception:
-                        try:
-                            return wkt.loads(ewkb)
-                        except Exception:
-                            pass
-    return None
+def _first_not_null_value(array):
+    return array.loc[~array.isnull()].iloc[0]
```

Comment:
What happens if they're all null? I think there will be an `IndexError`

Reply:
True that. We can rename the method and use an if everywhere it is used. cc @alrocar